src/mvesuvio/analysis_reduction.py

```diff
@@ -1,145 +1,197 @@
 import numpy as np 
 import matplotlib.pyplot as plt
 import scipy
+import dill      # Only for converting constraints from string
+from mantid.kernel import StringListValidator, Direction, IntArrayBoundedValidator, IntArrayProperty,\
+     IntBoundedValidator, FloatBoundedValidator
+from mantid.api import FileProperty, FileAction, PythonAlgorithm, MatrixWorkspaceProperty
+from mantid.dataobjects import TableWorkspaceProperty
 from mantid.simpleapi import mtd, CreateEmptyTableWorkspace, SumSpectra, \
                             CloneWorkspace, DeleteWorkspace, VesuvioCalculateGammaBackground, \
                             VesuvioCalculateMS, Scale, RenameWorkspace, Minus, CreateSampleShape, \
                             VesuvioThickness, Integration, Divide, Multiply, DeleteWorkspaces, \
-                            CreateWorkspace
+                            CreateWorkspace, CreateSampleWorkspace
 
 from mvesuvio.util.analysis_helpers import loadConstants, numericalThirdDerivative
 
-from dataclasses import dataclass
 
 
 np.set_printoptions(suppress=True, precision=4, linewidth=200)
 
 
-@dataclass(frozen=False)
-class NeutronComptonProfile:
-    label: str
-    mass: float
+class AnalysisRoutine(PythonAlgorithm):
```

Comment:
The name AnalysisRoutine is rather generic. Once added to Mantid, it won't be obvious that it belongs to the algorithms specific for Vesuvio. Maybe rename it to VesuvioAnalysisRoutine?